### DISCUSSION TITLE
h264 video cannot be played on IOS devices.

### DISCUSSION THREAD

**Discussion by huahuasousou** [photoprism/photoprism#2852] [Q&A]
After updating from photoprism/photoprism:220901-bullseye to the latest version, 100mbps h264 video files recorded by sony A7RM3 cannot be played on ios devices. I tried with iphone13pro and iphone13pro max, ipad pro 12.9 2018 and tested at the same time. None of them can play the file.
Meanwhile, chrome on ubuntu and windows plays normal.

Also, I'm sure I don't have ffmpeg reencoding turned on, since I'm deploying within a LAN, I don't need to wait for the transcoding to complete.

I tried converting the video from h264 to h264 again and that solved the problem. But I don't want to do this, it will cause extra storage space usage. And in previous versions there was no problem.

you can find the H264 video file in here:
https://drive.google.com/file/d/10ok-2uvjHqj_OBbhgsfbqgjo2y2qI_1-/view?usp=sharing

**Comment by graciousgrey** [comment]
As it is working in chrome, this is rather sounds like an issue with your device and not with PhotoPrism

**Reply by huahuasousou** [reply]
I have verified again that it is playing correctly in 220901-bullseye. So this problem can be reproduced, simply revert back to the 220901-bullseye version and it will work fine.

**Reply by lastzero** [reply]
Bullseye uses an old FFmpeg version. You're welcome to test different encoder versions and let us know if there's a difference. You can also enable trace mode to see more detailed logs when transcoding. However, please understand that we cannot spend much more time testing this on our own if you're not a PhotoPrism+ member or business customer.

**Reply by huahuasousou** [reply]
I'm not using ffmpeg function, I want to stream directly to the client. I set the conversion function of ffmpeg to start only when it is more than 500mbps, so it should not be activated.  And by looking at the debug output, it is directly stream on the PC platform is fine, but it cannot be play on the iOS device.  So this doesn't involve any code for ffmpeg convert. 

Please really test the video I have provided and you will find the problem.

In fact, there are many problems here. For example, photoprism have not correctly analysis the .mp4 file created by the sony camera, such as extract the metadata, and cannot get  the shooting date and camera model of the file. 

The Sony video can’t not play in new version, I hope it can be fixed, because in old version its fine.

**Reply by lastzero** [reply]
Be aware that due to the different Linux distributions ALL the dependencies may in fact be different, NOT JUST FFMPEG. That means, the detected codec could be different. We have also a new mime type detection, which could yield different results. If you have chosen to pay less than other users (or nothing), we have to ask you to support our efforts and help with testing. Thank you very much! ❤️
